darts/models/rnn_model.py

```diff
@@ -42,79 +45,64 @@ def __init__(self,
 
         # Defining parameters
-        self.hidden_dim = hidden_dim
-        self.n_layers = num_layers
         self.target_size = target_size
-        num_layers_out_fc = [] if num_layers_out_fc is None else num_layers_out_fc
-        self.out_len = output_chunk_length
         self.name = name
 
         # Defining the RNN module
-        self.rnn = getattr(nn, name)(input_size, hidden_dim, num_layers, batch_first=True, dropout=dropout)
-
-        # The RNN module is followed by a fully connected layer, which maps the last hidden layer
-        # to the output of desired length
-        last = hidden_dim
-        feats = []
-        for feature in num_layers_out_fc + [output_chunk_length * target_size]:
-            feats.append(nn.Linear(last, feature))
-            last = feature
-        self.fc = nn.Sequential(*feats)
-
-    def forward(self, x):
-        # data is of size (batch_size, input_chunk_length, input_size)
+        self.rnn = getattr(nn, name)(input_size, hidden_dim, 1, batch_first=True, dropout=dropout)
+
+        # The RNN module needs a linear layer V that transforms hidden states into outputs, individually
+        self.V = nn.Linear(hidden_dim, target_size)
+
+    def forward(self, x, h=None):
+        # data is of size (batch_size, input_length, input_size)
         batch_size = x.size(0)
 
-        out, hidden = self.rnn(x)
+        # out is of size (batch_size, input_length, hidden_dim)
+        out, last_hidden_state = self.rnn(x) if h is None else self.rnn(x, h)
+        # TODO: confirm shape of out
 
-        """ Here, we apply the FC network only on the last output point (at the last time step)
+        """ Here, we apply the V matrix to every hidden state to produce the outputs
         """
-        if self.name == "LSTM":
-            hidden = hidden[0]
-        predictions = hidden[-1, :, :]
-        predictions = self.fc(predictions)
-        predictions = predictions.view(batch_size, self.out_len, self.target_size)
+        # TODO: for one layer out should be equal to hidden, make sure that's true
+        predictions = self.V(out)  # TODO: make sure this matrix multiplication is correct
+
+        # predictions is of size (batch_size, input_length, target_size)
+        predictions = predictions.view(batch_size, -1, self.target_size)
 
-        # predictions is of size (batch_size, output_chunk_length, 1)
-        return predictions
+        # returns outputs for all inputs, only the last one is needed for prediction time
+        return predictions, last_hidden_state
 
 
 class RNNModel(TorchForecastingModel):
     @random_method
     def __init__(self,
-                 input_chunk_length: int,
-                 output_chunk_length: int,
                  model: Union[str, nn.Module] = 'RNN',
-                 hidden_size: int = 25,
-                 n_rnn_layers: int = 1,
-                 hidden_fc_sizes: Optional[List] = None,
+                 input_chunk_length: int = 12,
```

Comment:
[For later]: could we imagine having None here for recurrent models, which would mean something "use the longest sequence available"?

Reply:
Yes, I was also thinking about this. I created a corresponding Jira task. The main problem with removing the fixed length requirement is that we create tensors that contain several time series at once. So we would have to create torch tensors that contain time series with different lengths, which might be a bit tricky, but I am not sure.

Reply:
OK perfect, let's revisit at some point.